src/commands/CmdContinue.cpp

```diff
@@ -59,6 +60,34 @@ int CmdContinue (
 
     to_copy = tracked[tracked.size () - id];
   }
+  else if (!tags.empty())
```

Comment:
Do not implement tag filtering here, use the available on in `getTracked` (You need to create a filter with the given tag set). With that you should only need to pick the first interval from the returned array